### DISCUSSION TITLE
send OTP in email link from Supabase-js client

### DISCUSSION THREAD

**Discussion by zxqp** [supabase/supabase#5916] [Questions] [answered]
Referring to: [https://github.com/supabase/gotrue/releases/tag/v2.5.8](https://github.com/supabase/gotrue/releases/tag/v2.5.8)

How to use send OTP in email link from Supabase-js client?

**Comment by burggraf** [comment]
Is this what you're looking for?  [Using OTP with password based logins](https://supabase.com/docs/guides/auth/auth-twilio#using-otp-with-password-based-logins)

**Reply by zxqp** [reply]
That's OTP via phone, Im looking for OTP via email referring to the release note attached with the message

**Reply by burggraf** [reply]
As I understand it, OTP via email doesn't exist.  That would be a "magic link".  @kangmingtay may need to confirm this, but that's how I understand it.

**Reply by GaryAustin1** [reply]
I believe OTP via email is being worked on based on the original link here and the first part of that work being done is supabase/gotrue to support passing back a token/OTP.  In order for that to eventually work, would need I'm guessing a template fix in studio for that info in the email template and then gotrue.js (and other languages) to be updated to take a token and pass it back to the gotrue server. 
This is sort of the root of the discussion: https://github.com/supabase/gotrue/issues/264
No where have I seen a roadmap though, just the background issues and PR's.
Edit:  I'll also say I've been watching this as it is important for PWA webapps which I'm working on, as clicking on an email confirm link opens a new browser window, versus the user going back to my homescreen installed app and entering a code.

**Reply by zxqp** [reply]
+1 @GaryAustin1

**Comment by GaryAustin1** [comment, marked as answer]
I can confirm per the latest blog post: https://supabase.com/blog/2022/03/28/community-day (scroll to Send OTP via email)

Email OTP is working!

Email on signup with template modified with {{ .Token}}
![image](https://user-images.githubusercontent.com/54564956/160448771-a1f0b508-3688-4c32-a24a-8cc992e5db47.png)

And call made to confirm the user.
```
const {user,session,error} = await supabase.auth.verifyOTP ({
	email:'name@email.com',
	token:'owOWaODcyzG0ItuHQ24ipA',
	type:'signup'
})
```
